src/LuaTransformer.ts

```diff
@@ -3324,10 +3412,14 @@ export class LuaTransformer {
         }
     }
 
-    public transformClassExpression(expression: ts.ClassExpression): ExpressionVisitResult {
+    public transformClassExpression(expression: ts.ClassLikeDeclaration): ExpressionVisitResult {
+        const isDefaultExport = tsHelper.hasDefaultExportModifier(expression.modifiers);
+
         const className =
             expression.name !== undefined
                 ? this.transformIdentifier(expression.name)
+                : isDefaultExport
+                ? this.createDefaultExportIdentifier(expression)
```

Comment:
I don't like nested ternary conditions, or how this is formatted. In case of this nesting I'd expect an extra indent on the cases of the second conditional. But it might be nicer to refactor this to just an if statement or maybe another method.